ui/dashboards/src/components/Variables/TemplateVariable.tsx

```diff
@@ -107,6 +108,7 @@ function ListVariable({ name }: TemplateVariableProps) {
           label={title}
           value={selectValue}
           onChange={(e) => {
+            setVariableDefaultValue(name, e.target.value);
```

Comment:
I feel like it makes more sense if we only call this `onSave` and not on every change

Reply:
After refactoring and moving the updateVariableDefaultValues logic to the variable store, this isn't needed and is removed now!